### PR TITLE
fix(operator/do): fix typings

### PR DESCRIPTION
**Pull request by Brooooooklyn**
**Description:**
```ts
Observable.of('abc')
  .do(v => {
    // type of v is { } here
    console.log(v.substr(2))
  })
```

**Related issue (if exists):**

### CODE REVIEW

**Comment by kwonoj** [comment]
This looks ok to me, @david-driscoll is this caused by script running? or current state is expected?

**Comment by blesh** [comment]
This looks fine to me, pending @david-driscoll's approval (because it's typings related)

File: src/operator/do.ts [hunk 1]
```diff
@@ -48,7 +48,7 @@ import { TeardownLogic } from '../Subscription';
  * @owner Observable
  */
 /* tslint:disable:max-line-length */
-export function _do<T>(next: (x: T) => void, error?: (e: any) => void, complete?: (this: Observable<T>) => void): Observable<T>;
+export function _do<T>(this: Observable<T>, next: (x: T) => void, error?: (e: any) => void, complete?: (this: Observable<T>) => void): Observable<T>;
```
Comment (david-driscoll) [inline src/operator/do.ts]: definitely something I missed with the script.  This needs to be changed slightly, note the additional `this: Observable<T>` .

`complete?: (this: Observable<T>) => void` should be changed back to `complete?: () => void`

Comment (Brooooooklyn) [inline src/operator/do.ts]: fixed

**Comment by david-driscoll** [comment]
LGTM

**Comment by kwonoj** [comment]
LGTM

**Pull request merged**